google-cloud-firestore/lib/google/cloud/firestore/database.rb

```diff
@@ -0,0 +1,1013 @@
+            raise err
+          end
+        end
+
+        ##
+        # Create a read-only transaction to perform multiple reads that are
+        # executed at a single logical point in time in a database.
+        #
+        # Changes to data are not supported. See {ReadOnlyTransaction}.
+        #
+        # @param [Time] read_time The time to read the documents at. This may
+        #   not be older than 60 seconds. Optional.
+        #
+        # @yield [read_transaction] The block for reading data.
+        # @yieldparam [ReadOnlyTransaction] read_transaction The read-only
+        #   transaction object for reading data.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.read_only_transaction do |rtx|
+        #     # Create a query
+        #     query = rtx.query.select(:population).from(:cities)
+        #
+        #     # Get/run a query
+        #     rtx.get(query).each do |city|
+        #       puts "#{city.document_id} has #{city[:population]} residents."
+        #     end
+        #   end
+        #
+        def read_only_transaction read_time: nil
+          read_tx = ReadOnlyTransaction.from_database self, read_time: read_time
+          yield read_tx
+          read_tx.rollback
+        end
+        alias_method :read_transaction, :read_only_transaction
+        alias_method :snapshot, :read_only_transaction
+
+        ##
+        # Creates a query object.
+        #
+        # @return [Query] A query.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Create a query
+        #   query = firestore.query.select(:population).from(:cities)
+        #
+        #   # Get/run a query
+        #   firestore.get(query).each do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def query
+          Query.start "#{path}/documents", self
+        end
+        alias_method :q, :query
+
+        ##
+        # Creates a query object with method `select` called on it. (See
+        # {Query#select}.)
+        #
+        # @param [String, Symbol] fields A field mask to filter results with and
+        #   return only the specified fields. One or more field paths can be
+        #   specified.
+        #
+        # @return [Query] A query with `select` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Create a query
+        #   query = firestore.select(:population).from(:cities)
+        #
+        #   # Get/run a query
+        #   firestore.get(query).each do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def select *fields
```

Comment:
Sorry, but I really don't think we should be duplicating these comments multiple times. They are super helpful and well-written, but they will just be as useful if they only exist in a single parent class :)